ckan/lib/cli.py

```diff
@@ -625,6 +628,35 @@ def clear(self):
         else:
             clear(package_id)
 
+
+    def remove_orphans(self):
+        indexed_package_ids = []
+        rows = 100
+        while True:
+            search = logic.get_action('package_search')({},{
+                                'q': '*:*',
+                                'start': rows - 100,
+                                'rows': rows,
+                                'fl': 'id'})
+            indexed_package_ids += [r['id'] for r in search['results']]
+            if len(search['results']) == 0:
+                break
+            rows += 100
+
+        import ckan.model as model
+        from sqlalchemy.sql import select
+        package_ids = [r[0] for r in select([model.package_table.c['id']]).execute()]
```

Comment:
you can use a set here instead of a list for O(1) access times

Reply:
@wardi lists have been changed to sets